### PR TITLE
Add Changelog

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description
Add a changelog to keep track of changes for releases. Part of the PR process should be to add a release note to this.

I've back filled the changelog using the existing release notes and PRs in `0.1.1..HEAD`.

#### Tests
N/A

#### Documentation
Yes!

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+- Added the ability to acknowledge `AuthorityLossImminent` messages.
+- Added an `Open Inspector` button to the `SpatialOS` menu in the Unity Editor.
+- Added support for local mobile development
+- Added a changelog
+
+### Changed
+
+- Changed the allocation type used internally for Unity ECS chunk iteration from `Temp` to `TempJob`
+- Running a build in the Editor no longer automatically selects all scenes in the Unity build configuration
+- `Improbable.Gdk.Core.Snapshot::AddEntity` now returns the `EntityId` assigned in the snapshot.
```
Comment (jared-improbable) [inline CHANGELOG.md]: C++-ism `::` -> `.`

Comment (jamiebrynes7) [reply CHANGELOG.md]: 😬 Been writing too much Rust recently

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Changelog
+
+## Unreleased
+
+### Added
+
+- Added support for the Steam authentication flow.
+- Added the ability to acknowledge `AuthorityLossImminent` messages.
+- Added an `Open Inspector` button to the `SpatialOS` menu in the Unity Editor.
+- Added support for local mobile development
```
Comment (jessicafalk) [inline CHANGELOG.md]: full stop at the end is missing

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+- Added a changelog
```
Comment (jessicafalk) [inline CHANGELOG.md]: 10 / 10

**Pull request merged**